src/collar/oc_api.lsl

```diff
@@ -173,16 +173,13 @@ PrintAccess(key kID){
     //llSay(0, sFinal);
 }
 
-list g_lActiveListeners;
+integer g_iListener;
+integer g_iChatListener;
 DoListeners(){
-    integer i=0;
-    integer end = llGetListLength(g_lActiveListeners);
-    for(i=0;i<end;i++){
-        llListenRemove(llList2Integer(g_lActiveListeners, i));
-    }
-    
-    g_lActiveListeners = [llListen(g_iChannel, "","",""), llListen(0,"","",""),  llListen(g_iInterfaceChannel, "", "", "")];
-    
+    if (g_iListener) llListenRemove(g_iListener);
+    if (g_iChatListener) llListenRemove(g_iChatListener);
+    g_iListener = llListen(llAbs(g_iChannel), "","","");
```

Comment:
to listen to a private channel, a positive g_iChannel value is always used

Reply:
If someone tried to set a negative one, then silently switching that to positive seems like it would be confusing.

Reply:
ok, I agree, this method is not suitable, I will look for another

Reply:
ok, let's leave the negative channel unchanged